src/Analyzers/Core/Analyzers/SimplifyInterpolation/Helpers.cs

```diff
@@ -132,6 +136,59 @@ private static IOperation Unwrap(IOperation expression)
             formatString = null;
         }
 
+        private static bool IsObjectToStringOverride(IMethodSymbol method)
+        {
+            while (method.OverriddenMethod is not null)
+                method = method.OverriddenMethod;
+
+            return method.ContainingType.SpecialType == SpecialType.System_Object
+                && method.Name == nameof(ToString);
+        }
+
+        private static bool IsInvariantCultureReference(IOperation operation)
+        {
+            if (Unwrap(operation) is not IPropertyReferenceOperation { Member: { } member })
+                return false;
+
+            return member.Name switch
+            {
+                nameof(CultureInfo.InvariantCulture) => SymbolEqualityComparer.Default.Equals(
+                    member.ContainingType,
+                    operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.CultureInfo).FullName!)),
+
+                "InvariantInfo" =>
+                    SymbolEqualityComparer.Default.Equals(
+                        member.ContainingType,
+                        operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.NumberFormatInfo).FullName!))
+                    || SymbolEqualityComparer.Default.Equals(
+                        member.ContainingType,
+                        operation.SemanticModel!.Compilation.GetTypeByMetadataName(typeof(System.Globalization.DateTimeFormatInfo).FullName!)),
+
+                _ => false,
```

Comment:
i honestly find this siwtch hard to read. can you break into separate ops?

Reply:
Meaning separate `if` statements, or what did you have in mind?